File: IN1010/Obliger/Oblig1/celle.java
public class Celle {
    boolean levende = false;
    Celle[] naboer = new Celle[8];
    int antNaboer = 0;
    int antLevendeNaboer = 0; //konstruktøren til Celle

    public void settDoed() {
        levende = false;
    }

    public void settLevende() {
        levende = true;
    }

    public boolean erLevende() {
        return levende;
    }

    public char hentStatusTegn() { //returner en char som er bare en enkel bokstav
        if (levende) {
            return 'O';
        } return '.';
    }

    public void leggTilNabo(Celle nabo) {
        for (int i = 0; i < naboer.length; i++) { //looper igjennom arrayen for å finne en ledig plass
            if (naboer[i] == null) {
                naboer[i] = nabo;
                break; 
            }
        } 
        antNaboer = 0;
        for (Celle celle : naboer) { //teller antall ikke null-objekter
            if (celle != null) {
                antNaboer++;
            }
        }
    }
    
    public void tellLevendeNaboer() { //går igjennom arrayen for å sjekke om de er levende eller ikke og teller
        int antLevende = 0;
        for (Celle nabo : naboer) {
            if (nabo != null) {
                if (nabo.erLevende()) {
                    antLevende++;
                }
            }
        }
        antLevendeNaboer = antLevende;
    }

    public void oppdaterStatus() { //en if sjekk med en logikk som sjekker om cellen skal være dø eller levende
        if (levende && (antLevendeNaboer < 2 || antLevendeNaboer > 3)) {
            levende = false;
        } else if (!levende && antLevendeNaboer == 3) {
            levende = true;
        }
    }
}

File: IN1000/Obliger/Uke 10 og 11 vanlig redo/celle.py
class Celle:
    # Konstruktør
    def __init__(self):
        self._status = "doed"
        self._naboer = []
        self._ant_levende_naboer = 0
    
    def sett_doed(self):
        self._status = "doed"

    def sett_levende(self):
        self._status = "levende"

    def er_levende(self):
        return self._status == "levende"

    def hent_status_tegn(self):
        if self.er_levende():
            return "O"
        return "."

    def legg_til_nabo(self, nabo):
        self._naboer.append(nabo)

    def tell_levende_naboer(self):
        self._ant_levende_naboer = 0
        for celle in self._naboer:
            if celle.er_levende():
                self._ant_levende_naboer += 1

    def oppdater_status(self):
        if self.er_levende():
            if self._ant_levende_naboer < 2:
                self.sett_doed()
            if self._ant_levende_naboer > 3:
                self.sett_doed()
        else:
            if self._ant_levende_naboer == 3:
                self.sett_levende()
    
    def __str__(self):
        return self.hent_status_tegn()
    